Server/src/main/java/org/xdi/oxauth/authorize/ws/rs/AuthorizeRestWebServiceImpl.java

```diff
@@ -187,7 +187,20 @@ public Response requestAuthorization(
         try {
             Map<String, String> customResponseHeaders = Util.jsonObjectArrayStringAsMap(customRespHeaders);
 
-            sessionIdService.assertAuthenticatedSessionCorrespondsToNewRequest(sessionUser, acrValuesStr);
+            try{
+                sessionIdService.assertAuthenticatedSessionCorrespondsToNewRequest(sessionUser, acrValuesStr);
+            }
+            catch (AcrChangedException e) { // Acr changed
+                //See https://github.com/GluuFederation/oxTrust/issues/797
+                if (e.isMethodEnabled()){
+                    if (!prompts.contains(Prompt.LOGIN)) {
+                        log.info("ACR is changed, adding prompt=login to prompts");
+                        prompts.add(Prompt.LOGIN);
```

Comment:
@jgomer2001 we added `login` to prompts array but passing `prompt` into `generateAuthenticatedSessionId()` method. It would mean that session has outdated value which is wrong.